### PR TITLE
Add benchmark using mu-haskell

### LINKED ISSUE #21: Add Haskell benchmark
Add a Haskell grpc benchmark.

The library is here: https://github.com/awakesecurity/gRPC-haskell

- [ ] gRPC server implementation
- [ ] Dockerfile to build an image with the gRPC server
- [ ] update build.sh, clean.sh and bench.sh scripts

### PR DESCRIPTION
**Pull request by akshaymankar**
The library is here: https://github.com/higherkindness/mu-haskell/

Fixes #21

### CODE REVIEW

**Comment by LesnyRumcajs** [comment]
@akshaymankar thank you for the contribution! Is the implementation going to work on multiple cores as well (and taking advantage of them)? I believe you should somehow instruct Haskell to grab only as many threads as there are CPUs intended for the benchmark, which is defined in `GRPC_SERVER_CPUS` environmental variable. See https://github.com/LesnyRumcajs/grpc_bench/pull/121 for a similar issue for Rust.

If it's intended only for single CPU then there's no issue and we can merge this without changes.

**Comment by akshaymankar** [comment]
Thanks for the pointer @LesnyRumcajs, I added a script to wrap the executable so the `GRPC_SERVER_CPUS` variable can be respected. Unfortunately, this is a config for the haskell runtime, so it cannot be put in the code.

File: haskell_mu_bench/run.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
```
Comment (LesnyRumcajs) [inline haskell_mu_bench/run.sh:3]: would the result be the same if we omitted this command and went ahead with just `sh` instead of `bash`?

**Review by LesnyRumcajs — APPROVED** [review]

File: haskell_mu_bench/run.sh (hunk 1 above)
Comment (akshaymankar) [reply haskell_mu_bench/run.sh:3]: Yeah, it would be the same. I can change it.

Comment (LesnyRumcajs) [reply haskell_mu_bench/run.sh]: Thank you, I like it minimal. :)

**Pull request merged**